framework/include/meshlet_helpers.hpp

```diff
@@ -0,0 +1,174 @@
+	 *								Additionally it provides the model and an optional mesh index if more data is needed. If no mesh index is provided then the meshes were combined beforehand.
+	 *								Must not grab ownership of the model, the model will be assigned to the meshlets after its execution.\n\n
+	 *								The callback has a specific layout, optional parameters can be omitted, but all of them need to be provided in the following order.
+	 *								Parameters in definition order:
+	 *								 - const std::vector<glm::vec3>& tVertices:		optional	The vertices of the mesh or combined meshes of the model
+	 *								 - const std::vector<uint32_t>& tIndices:  		mandatory	The indices of the mesh or combined meshes of the model
+	 *								 - const model_t& tModel:						mandatory	The model these meshlets are generated from
+	 *								 - std::optional<mesh_index_t> tMeshIndex: 		mandatory	The optional mesh index. If no value is provided, it means the meshes of the model are combined into a single vertex and index buffer
+	 *								 - uint32_t tMaxVertices:						mandatory	The maximum number of vertices that are allowed in a single meshlet.
+	 *								 - uint32_t tMaxIndices:						mandatory	The maximum number of indices that are allowed in a single meshlet.
+	 *								Return value:
+	 *								 - std::vector<meshlet>										The generated meshlets.\n\n
+	 *								Example:
+	 *								@code
+	 *								[](const std::vector<glm::vec3>& tVertices, const std::vector<uint32_t>& aIndices,
+	 *											const model_t& aModel, std::optional<mesh_index_t> aMeshIndex,
+	 *											uint32_t aMaxVertices, uint32_t aMaxIndices) {
+	 *									std::vector<gvk::meshlet> generatedMeshlets;
+	 *									// Do your meshlet division here
+	 *									return generatedMeshlets;
+	 *								}
+	 *								@endcode
+	 *	@param	aMaxVertices		The maximum number of vertices of a meshlet.
+	 *	@param	aMaxIndices			The maximum number of indices of a meshlet.
+	 *	@param	aCombineSubmeshes	If submeshes should be combined into a single vertex/index buffer.
+	 */
+	template <typename F>
+	extern std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModelsAndMeshletIndices, F aMeshletDivision,
+		const uint32_t aMaxVertices = 64, const uint32_t aMaxIndices = 378, const bool aCombineSubmeshes = true)
+	{
+		std::vector<meshlet> meshlets;
+		for (auto& pair : aModelsAndMeshletIndices) {
+			const auto& model = std::get<avk::resource_ownership<model_t>>(pair);
+			auto& meshIndices = std::get<std::vector<mesh_index_t>>(pair);
+
+			if (aCombineSubmeshes) {
+				auto [vertices, indices] = get_vertices_and_indices(std::vector({ std::make_tuple(avk::const_referenced(model.get()), meshIndices) }));
+				std::vector<meshlet> tmpMeshlets = divide_indexed_geometry_into_meshlets(vertices, indices, std::move(model), std::nullopt, aMaxVertices, aMaxIndices, std::move(aMeshletDivision));
```

Comment:
Hmm, does the compiler not care that `divide_indexed_geometry_into_meshlets` is declared and defined _after_ `divide_into_meshlets`? Usually, this leads to problems, but could be that it is a bit differently with templates. Anyways, why not just move  `divide_indexed_geometry_into_meshlets` above `divide_into_meshlets`?!

Reply:
huh good catch, appently it works. I'll move it up.